packages/backend/src/nest/libp2p/libp2p.auth.ts

```diff
@@ -414,6 +414,25 @@ export class Libp2pAuth {
       this.logger.warn(`Auth connection with ${peerId.toString()} was disconnected`)
       this.closeAuthConnection(peerId, false)
     }
+
+    if (this.joinStatus === JoinStatus.JOINED) {
+      return
+    }
+
+    /**
+     * We need to manually reset the join status in the case where the status is stuck on an intermediate state like
+     * JOINING when disconnecting (for example this can happen when the user you are connecting to doesn't have your
+     * information in their chain yet resulting in an invalid device error)
+     */
+    const oldJoinStatus = this.joinStatus
+    if (this.sigChainService.team == null) {
```

Comment:
I believe this should always be true or it will cause an assert error.
```suggestion
    if (this.sigChainService.activeChain?.team == null) {
```